src/nvim/api/vim.h

```diff
@@ -4,6 +4,10 @@
 #include <stdint.h>
 
 #include "nvim/api/private/defs.h"
+#include "nvim/lib/kvec.h"
+
+EXTERN kvec_t(String) namespaces INIT(= KVEC_INIT);
+EXTERN uint64_t next_namespace_id INIT(= 1);
```

Comment:
Do you plan on removing namespaces? Otherwise it is pointless to have `next_namespace_id` with the current implementation: you can just use `kv_size(namespaces)` after `kv_push`.